superset/views/core.py

```diff
@@ -832,7 +834,65 @@ def addtodashboard(self):
             return json_success('OK')
         except Exception as e:
             logging.exception(e)
-            return json_error_response(e)  
+            return json_error_response(e) 
+
+    def update_dashboard(self, dashboards, parameters):
+        dashboard = dashboards['dashboards'][0]
+        title = parameters['dashboard_title']
+        if title:
+            dashboard.dashboard_title = title
+        # assuming sinle dashboard per JSON
+        for slice_item in dashboard.slices:
+            datasource = parameters[slice_item.datasource_name]
+            if datasource is not None:
+                slice_item.alter_params(
+                            datasource_name = datasource.name,
+                            schema = datasource.name,
+                            database_name = datasource.database_name,
+                        )
+        return dashboards
+    
+
+    @expose('/replicate_dashboard', methods=['POST'])
+    def replicate_dashboard(self):
+        database_name = request.form.get('database_name')
+        sqlalchemy_uri = request.form.get('sqlalchemy_uri')
+        extra = request.form.get('extra')
+        impersonate_user = eval(request.form.get('impersonate_user'))
+        if database_name is not None:
+            database_id = create_database(database_name,sqlalchemy_uri,extra,impersonate_user)
+
+        dashboard_title = request.form.get('dashboard_title')
+        tablesparam = json.loads(r''+request.form.get('tables'))
```

Comment:
Please replace `tablesparam` with `tables_param` to make it more readable.

Reply:
Sure will do